Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -34,24 +37,35 @@ namespace Terrain
 
     TerrainMeshManager::TerrainMeshManager()
     {   
-        AzFramework::Terrain::TerrainDataNotificationBus::Handler::BusConnect();
     }
 
     TerrainMeshManager::~TerrainMeshManager()
-    {   
+    {
         AzFramework::Terrain::TerrainDataNotificationBus::Handler::BusDisconnect();
     }
 
-    void TerrainMeshManager::Initialize()
+    void TerrainMeshManager::Initialize(AZ::RPI::Scene& parentScene)
     {
-        if (!InitializeSectorModel())
+        AzFramework::Terrain::TerrainDataNotificationBus::Handler::BusConnect();
+
+        m_parentScene = &parentScene;
+
+        bool success = true;
+        success = success && InitializeCommonSectorData();
+        success = success && InitializeDefaultSectorModel();
+        if (!success)
         {
             AZ_Error(TerrainMeshManagerName, false, "Failed to create Terrain render buffers!");
             return;
         }
```

Comment:
Shouldn't the BusConnect() move _after_ the check for success? It seems like whatever reason you had for moving it from the constructor to here is probably the same reason you wouldn't want to connect to it in the failure case.